packages/react-instantsearch/src/core/utils.js

```diff
@@ -58,3 +58,6 @@ export function assertFacetDefined(searchParameters, searchResults, facet) {
 export function getDisplayName(Component) {
   return Component.displayName || Component.name || 'UnknownComponent';
 }
+
+const resolved = typeof Promise !== 'undefined' && Promise.resolve();
+export const defer = resolved ? f => { resolved.then(f); } : setTimeout;
```

Comment:
It uses the preact implementation for doing stuff async as soon as possible. The guy even publish a benchmark of the different options.
See: https://github.com/developit/preact/commit/6954d7549f35b5c32421c6855ee29ffda01137d1 and https://esbench.com/bench/55d4d44e2efbca1100bf7251